src/index.js

```diff
@@ -52,6 +58,19 @@ export default class AutoCheckElement extends HTMLElement {
     this.setAttribute('csrf', value)
   }
 
+  get required(): boolean {
+    return this.hasAttribute('required')
+  }
+
+  set required(required: boolean) {
+    this.input.required = required
+    if (required) {
+      this.input.setCustomValidity('auto-check-loading')
```

Comment:
Firefox shows this text in a tooltip, so maybe "Validating…" for the loading state and use the server's error message for the error state.